tests/integration/test_gui.py

```diff
@@ -32,7 +32,7 @@
 from tests.lib.logger import logger
 from tests.lib.fixtures import fixtures
 from tests.lib.pipes import push_event, push_events, uinput_write_history_pipe
-from tests.integration.test_components import FlowBoxTestUtils
+from tests.integration.test_components import FlowBoxTestUtils, GtkKeyEvent
```

Comment:
To debug the failing test_combination test I went through your changes and reverted them one by one locally, replacing the GtkKeyEvent class from this file with the import causes the test to fail somehow.

Reply:
oh, funny, now it doesn't fix it anymore.

Reply:
because it is a race condition. it fails most of the time, but works sometimes. This could be timing related. Looking at the test, there are some calls to `self.throttle`. Something is possibly happening slower than before.

Reply:
btw. I seem to be the only user of the class. Did it serve a purpose before?

Reply:
Also I found this:
```
    def throttle(self, time_=10):
        """Give GTK some time in ms to process everything."""
        # tests suddenly started to freeze my computer up completely and tests started
        # to fail. By using this (and by optimizing some redundant calls in the gui) it
        # worked again. EDIT: Might have been caused by my broken/bloated ssd. I'll
        # keep it in some places, since it did make the tests more reliable after all.
        for _ in range(time_ // 2):
            gtk_iteration()
            time.sleep(0.002)
```
I wonder if we can use sth. like `GLib.timeout_add()` instead. From what I know `time.sleep` is discouraged for managing any kind of timing issues. Always try to wait for the desired state to appear, rather than waiting fixed amounts of time. It is just a guess, but `time.sleep` may actually prevent other critical code from being executed during the sleep (on the main thread I guess, but I do not know how Python + Gtk works with regards to UI thread vs. background threads) 🤷🏼‍♂️ 

I will try to add sth. like this:

```
        self.awaitEqual(
            lambda t: self.data_manager.active_preset.get_mapping(
                get_combination(combination_1)
            ).output_symbol,
            "a",
            timeout=1000,  # optional, default maybe at 100
        )
        self.awaitTrue(
            # ...
        )
```

Reply:
I think our tests are horribly complicated. I can also see that `throttle` might be a bad practice.

`test_combination` works when my current PR #578 is merged with yours, so don't worry about it too much

Reply:
#578 is merged, once you pull the newest changes from beta it should work.

Reply:
I resolved the conflicts. Yes the TestGui suite now runs fine 👍🏼.